.github/workflows/remove-wheels.yml

```diff
@@ -55,19 +57,36 @@ jobs:
 
       - name: Remove old uploads to save space
         run: |
-          # Remove all _but_ the last ${N_LATEST_UPLOADS} package versions
-          # N.B.: `anaconda show` places the newest packages at the bottom of the output
-          # of the 'Versions' section and package versions are preceded with a '   + '.
+          # Remove all _but_ the last ${N_LATEST_UPLOADS} package versions and
+          # remove all package versions older than 30 days.
 
           if [ -s package-names.txt ]; then
+              threshold_date="$(date +%F -d '30 days ago')"
+
               # Remember can't quote subshell as need to split on (space seperated) token
               for package_name in $(cat package-names.txt); do
 
                   echo -e "\n# package: ${package_name}"
 
-                  anaconda show "${ANACONDA_USER}/${package_name}" &> >(grep '+') | \
-                      awk '{print $2}' | \
-                      head --lines "-${N_LATEST_UPLOADS}" > remove-package-versions.txt
+                  curl --silent https://api.anaconda.org/package/"${ANACONDA_USER}/${package_name}" | \
+                      jq -r '.releases[].version' > package-versions.txt
+                  head --lines "-${N_LATEST_UPLOADS}" package-versions.txt > remove-package-versions.txt
+
+                  # Don't need to check for duplicate versions added to removal
+                  # given nightly CRON basis.
+                  for package_version in $(cat package-versions.txt); do
```

Comment:
Very much a corner case, but could it still produce a duplicate when the No6 is just crosses the 30-day threshold? 

(but then, if removing a non-existing package in the removal loop below, then this is not a problem at all)